storage/postgres/log_storage.go

```diff
@@ -502,16 +502,16 @@ func (t *logTreeTX) QueueLeaves(ctx context.Context, leaves []*trillian.LogLeaf,
 		}
 
 		// Create the work queue entry
+		queueTimestamp, err := ptypes.Timestamp(leaf.QueueTimestamp)
+		if err != nil {
+			return nil, fmt.Errorf("got invalid queue timestamp: %v", err)
+		}
 		args := []interface{}{
 			t.treeID,
 			leaf.LeafIdentityHash,
 			leaf.MerkleLeafHash,
+			queueTimestamp.UnixNano(),
 		}
```

Comment:
Much better :slightly_smiling_face:

Reply:
Except it broke for number of params, unexpected as I thought queueArgs returned one value.

Reply:
OK mystery solved - caused by build tag for batch queue. So it has to stay for now I think. Will update.